atc/creds/credhub/credhub.go

```diff
@@ -18,12 +18,15 @@ type CredHubAtc struct {
 }
 
 // NewSecretLookupPaths defines how variables will be searched in the underlying secret manager
-func (c CredHubAtc) NewSecretLookupPaths(teamName string, pipelineName string) []creds.SecretLookupPath {
+func (c CredHubAtc) NewSecretLookupPaths(teamName string, pipelineName string, allowRootPath bool) []creds.SecretLookupPath {
 	lookupPaths := []creds.SecretLookupPath{}
 	if len(pipelineName) > 0 {
 		lookupPaths = append(lookupPaths, creds.NewSecretLookupWithPrefix(path.Join(c.prefix, teamName, pipelineName)+"/"))
 	}
 	lookupPaths = append(lookupPaths, creds.NewSecretLookupWithPrefix(path.Join(c.prefix, teamName)+"/"))
+	if allowRootPath {
```

Comment:
@vito WRT [concourse/rfcs#21 (comment)](https://github.com/concourse/rfcs/pull/21#discussion_r269687922), I think that makes sense for team level credential manager, because that would be configured by concourse admin; but doesn't make that much sense at pipeline level, because it adds overhead for pipeline author, and hurts mobility of pipelines, consider if a pipeline yml is shared by multiple teams or multiple pipelines, or migrate a pipeline from one concourse instance to the other.

For us, I'd keep current `NewSecretLookupPaths` pattern. As Concourse doesn't support sub-team, a big BU may have multiple teams on our Concourse cluster, but the BU has a single Vault, so each team will still put their secrets under their own Vault path. At Vault side, permission management is the other headache, who can access which path, which is complex. Also given the nature of Security, migrating secrets of Vault, from one path to other path, from one Vault instance to other instance is not easy, not much tool can help. So I'd be very careful for any Vault related changes. As @ralekseenkov mentioned, it should not be any breaking changes.